public/app/features/search/state/SearchStateManager.ts

```diff
@@ -105,7 +105,10 @@ export class SearchStateManager extends StateManagerBase<SearchState> {
   };
 
   onStarredFilterChange = (e: FormEvent<HTMLInputElement>) => {
-    const starred = (e.target as HTMLInputElement).checked;
+    if (!(e.target instanceof HTMLInputElement)) {
+      return;
+    }
+    const starred = e.target.checked;
```

Comment:
why don't we just use the following?  
```ts
const starred = e.currentTarget.checked;
```
We've already declared the type in generic. Am I missing something?